packages/main/src/SplitButton.ts

```diff
@@ -285,22 +270,26 @@ class SplitButton extends UI5Element {
 
 		this._shiftOrEscapePressed = false;
 		this._setTabIndexValue();
-		this._manageFocus();
 	}
 
 	_onFocusIn(e: FocusEvent) {
 		if (this.disabled || getEventMark(e)) {
 			return;
 		}
 		this._shiftOrEscapePressed = false;
-		this._manageFocus(this);
+		this.focus();
 	}
 
 	_textButtonFocusIn(e?: FocusEvent) {
 		e?.stopPropagation();
-		this._manageFocus(this.textButton!);
+		this._setTabIndexValue(true);
+		this.textButton!.focus();
+	}
 
-		this._setTabIndexValue();
+	_arrowButtonFocusIn(e?: FocusEvent) {
+		e?.stopPropagation();
+		this._setTabIndexValue(true);
+		this.arrowButton!.focus();
 	}
```

Comment:
If these two are the same, can't we combine them into one handler? Check if the FocusEvent provides the element we're focusing and if so, I think combing them will be good. If not, we can still exctract most of it in a method as 80% of the logic is the same.